ontolearner/learner/term_typing/sbunlp.py

```diff
@@ -0,0 +1,400 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, List, Optional
+import re
+
+import torch
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+from ...base import AutoLearner
+
+class SBUNLPZSLearner(AutoLearner):
+    """
+    Qwen-based blind term typing learner (Task B), implemented as an AutoLearner.
+
+    This class reproduces the notebook logic:
+      - Fit phase learns the *allowed type inventory* from training data.
+      - Predict phase performs blind prompting per term using the learned type list.
+      - Outputs are restricted to the allowed types and returned as [{"id", "types"}].
+
+    Expected I/O (recommended):
+      - fit(train_data, task="term-typing", ontologizer=True):
+          The framework's AutoLearner.tasks_data_former() provides a unique list of
+          type labels; we store it to `self.allowed_types`.
+      - predict(eval_data, task="term-typing", ontologizer=False):
+          Pass a list of dicts with keys {"id": str, "term": str} so IDs are preserved.
+          Returns a list of dicts [{"id": ..., "types": [...] }].
+    """
+
+    def __init__(
+        self,
+        model_id: str = "Qwen/Qwen2.5-0.5B-Instruct",
+        device: Optional[str] = None,
+        max_new_tokens: int = 64,
+        temperature: float = 0.0,
+        token: Optional[str] = None,
+    ) -> None:
+        """
+        Args:
+            model_id: HF model id for Qwen.
+            device: "cuda", "mps", or "cpu". Auto-detected if None.
+            max_new_tokens: Generation cap per prompt.
+            temperature: Not used for greedy decoding (kept for future).
+            token: HF token if the model is gated.
+        """
+        super().__init__()
+
+        # Basic configuration
+        self.model_id = model_id
+        # default device detection: prefer CUDA if available
+        self.device = device or ("cuda" if torch.cuda.is_available() else "cpu")
+        self.max_new_tokens = max_new_tokens
+        self.temperature = temperature
+        self.token = token
+
+        # Model/tokenizer placeholders (populated by load())
+        self.tokenizer: Optional[AutoTokenizer] = None
+        self.model: Optional[AutoModelForCausalLM] = None
+
+        # Learned inventory of allowed type labels (populated by fit())
+        self.allowed_types: List[str] = []
+
+        # Regex used to extract quoted strings from model output (e.g. "type")
+        self._quoted_re = re.compile(r'"([^"]+)"')
+
+    def load(self, **kwargs: Any):
+        """
+        Load Qwen model and tokenizer.
+
+        NOTE:
+          - The HF arguments used here mirror your original code (`token=...`).
+            You may see a deprecation warning for `torch_dtype` (older transformers);
+            switching to `dtype=` is recommended but I did not change behavior here.
+        """
+        # Respect overrides from kwargs if provided
```

Comment:
but you are passing such information (model_id and token) inside the __init__ function. it would be great if this beeing clarified! i would say lets move them here instead of init.

Reply:
also use model_id and token names in args instead of **kwargs. Anyway check the parent class to adapt these changes.